nodl_python/nodl/schemas/v1.xsd

```diff
@@ -1,73 +1,62 @@
+            <xs:enumeration value="client" />
+            <xs:enumeration value="both" />
+        </xs:restriction>
+    </xs:simpleType>
 
-    <xs:element name="qos">
-        <xs:complexType>
-            <xs:attribute name="history" type="xs:string"/>
-            <xs:attribute name="reliability" type="xs:string"/>
-            <xs:attribute name="durability" type="xs:string"/>
-            <xs:attribute name="depth" type="xs:integer"/>
-            <xs:attribute name="lifespan" type="xs:decimal"/>
-            <xs:attribute name="deadline" type="xs:decimal"/>
-            <xs:attribute name="liveliness" type="xs:string"/>
-            <xs:attribute name="liveliness_lease_duration" type="xs:decimal"/>
-            <xs:attribute name="avoid_ros_namespace_conventions" type="xs:boolean"/>
-        </xs:complexType>
-    </xs:element>
+    <xs:simpleType name="pubSubType">
+        <xs:restriction base="xs:string">
+            <xs:enumeration value="publisher" />
+            <xs:enumeration value="subscription" />
+            <xs:enumeration value="both" />
+        </xs:restriction>
+    </xs:simpleType>
 
     <xs:element name="action">
         <xs:complexType>
-            <xs:sequence>
-                <xs:element ref="qos"/>
-            </xs:sequence>
-            <xs:attribute name="name" type="xs:string" use="required"/>
-            <xs:attribute name="type" type="xs:string" use="required"/>
-            <xs:attribute name="server" type="xs:boolean" default="false"/>
-            <xs:attribute name="client" type="xs:boolean" default="false"/>
+            <xs:attribute name="name" type="xs:string" use="required" />
+            <xs:attribute name="type" type="xs:string" use="required" />
+            <xs:attribute name="direction" type="serverClientType" use="required" />
         </xs:complexType>
     </xs:element>
 
     <xs:element name="parameter">
         <xs:complexType>
-            <xs:attribute name="name" type="xs:string" use="required"/>
-            <xs:attribute name="type" type="xs:string" use="required"/>
+            <xs:attribute name="name" type="xs:string" use="required" />
+            <xs:attribute name="type" type="xs:string" use="required" />
         </xs:complexType>
     </xs:element>
 
     <xs:element name="service">
         <xs:complexType>
-            <xs:sequence>
-                <xs:element ref="qos"/>
-            </xs:sequence>
-            <xs:attribute name="name" type="xs:string" use="required"/>
-            <xs:attribute name="type" type="xs:string" use="required"/>
-            <xs:attribute name="server" type="xs:boolean" default="false"/>
-            <xs:attribute name="client" type="xs:boolean" default="false"/>
+            <xs:attribute name="name" type="xs:string" use="required" />
+            <xs:attribute name="type" type="xs:string" use="required" />
+            <xs:attribute name="direction" type="serverClientType" use="required" />
```

Comment:
I'm not sold on the use of "direction". None of these are actually directions (server, client, publisher, subscription). Would "role" be better?

Reply:
I like role, changing it

Reply:
Changed in 42a43ab